src/event_form/components/calendar/base-calendar.controller.js

```diff
@@ -176,6 +177,11 @@ function BaseCalendarController(calendar, $scope) {
             (timeSpan.start && timeSpan.end) &&
             moment(timeSpan.start).isSame(timeSpan.end, 'day') &&
             moment(timeSpan.start).isAfter(timeSpan.end);
+      },
+      'tooFarInFuture': function (timespan) {
+        var d = new Date();
+        var maxDate = new Date(d.getFullYear() + calendar.maxYearTimeSpan, d.getMonth(), d.getDate());
+        return moment(timeSpan.start).isAfter(maxDate) || moment(timeSpan.end).isAfter(maxDate) ;
```

Comment:
can we leverage the utility of `momentjs` here?
```
var maxdate = moment(new Date()).add(calendar.maxYearTimeSpan, 'y');
return moment(timeSpan.start).isAfter(maxDate) || moment(timeSpan.end).isAfter(maxDate);
```

Do we need the first check on the `timeSpan.start`? the `timeSpan.end` is always going to be later, so it's safe to only check that property, no?

I think it could just be:

```
return moment(timeSpan.end).isAfter(moment().add(calendar.maxYearTimeSpan, 'y'));
```

Reply:
@jdrieghe 
Both very valid suggestions 👍 .
I've updated the PR.